index.bs

```diff
@@ -571,32 +572,31 @@ partial interface NeuralNetworkContext {
         let cellInput = nn.squeeze(nn.slice(input, [slice, 0, 0], [slice + 1, -1, -1]), [0]);
 
         let result = nn.gruCell(
-          hiddenSize, cellInput, cellWeight[slot], cellRecurrentWeight[slot],
-          cellHidden[slot], cellBias[slot], cellRecurrentBias[slot], 
-          resetAfter, layout, activations);
+          cellInput, cellWeight[slot], cellRecurrentWeight[slot], cellHidden[slot], 
+          hiddenSize, cellBias[slot], cellRecurrentBias[slot], resetAfter, layout, activations);
 
         cellOutput = (cellOutput ? nn.concat([cellOutput, result], 0) : result);
       }
 
-      hidden = cellOutput;
+      hiddenState = cellOutput;
 
       if (returnSequence) {
         cellOutput = nn.reshape(cellOutput, [1, slots, -1, hiddenSize]);
         sequence = (sequence ? nn.concat([sequence, cellOutput], 0) : cellOutput);
       }
     }
 
-    return (sequence ? [hidden, sequence] : [hidden]);
+    return (sequence ? [hiddenState, sequence] : [hiddenState]);
     </pre>
     </div>
 </div>
 
 ### gruCell ### {#api-neuralnetworkcontext-grucell}
-A single time step of the Gated Recurrent Unit (GRU) recurrent network using an update gate and a reset gate to compute the hidden state that rolls into the output across the temporal sequence of a recurrent network.
+A single time step of the Gated Recurrent Unit [[GRU]] recurrent network using an update gate and a reset gate to compute the hidden state that rolls into the output across the temporal sequence of a recurrent network.
 <script type=idl>
 partial interface NeuralNetworkContext {
-  Operand gruCell(long hiddenSize, Operand input, Operand weight, 
-                  Operand recurrentWeight, Operand hidden,
+  Operand gruCell(Operand input, Operand weight,
+                  Operand recurrentWeight, Operand hiddenState, long hiddenSize, 
```

Comment:
As you use `initialHiddenState` in `gru`, would you like to use `initialHiddenState` for `gruCell`?

Reply:
`gruCell` only handles a single time step that accepts just a single input hidden state and produces just a single output hidden state while `gru` runs through the cell for the entire temporal sequence with multiple hidden states produced for and accepted by the cell in each time step. That's why the prefix *initial* is only applicable to `gru` and not `gruCell`. Let me know if that makes sense.

Reply:
It makes sense. Thanks for the explanation.